sklearn/decomposition/tests/test_nmf.py

```diff
@@ -99,11 +106,11 @@ def test_initialize_variants():
         assert_almost_equal(evl[ref != 0], ref[ref != 0])
 
 
-# ignore UserWarning raised when both solver='mu' and init='nndsvd'
 @ignore_warnings(category=UserWarning)
-@pytest.mark.parametrize('solver', ('cd', 'mu'))
-@pytest.mark.parametrize('init',
-                         (None, 'nndsvd', 'nndsvda', 'nndsvdar', 'random'))
+@pytest.mark.parametrize(['solver', 'init'],
+                         [['cd', None], ['cd', 'nndsvd'], ['cd', 'nndsvda'],
+                          ['cd', 'nndsvdar'], ['mu', None], ['mu', 'nndsvda'],
+                          ['mu', 'nndsvdar']])
```

Comment:
You removed the "random" init.
Note that you can also use previous formulation, and add `if solver == 'mu' and init == 'nndsvd': return`.